drivers/sensor/bmi270/bmi270_trigger.c

```diff
@@ -0,0 +1,309 @@
+	const struct bmi270_config *cfg = dev->config;
+	struct bmi270_data *data = dev->data;
+	int ret;
+
+#if CONFIG_BMI270_TRIGGER_OWN_THREAD
+	k_sem_init(&data->trig_sem, 0, 1);
+	k_thread_create(&data->thread, data->thread_stack, CONFIG_BMI270_THREAD_STACK_SIZE,
+			(k_thread_entry_t)bmi270_thread, data, NULL, NULL,
+			K_PRIO_COOP(CONFIG_BMI270_THREAD_PRIORITY), 0, K_NO_WAIT);
+#elif CONFIG_BMI270_TRIGGER_GLOBAL_THREAD
+	k_work_init(&data->trig_work, bmi270_trig_work_cb)
+#endif
+
+	ret = bmi270_init_int_pin(&cfg->int1, &data->int1_cb,
+				  bmi270_int1_callback);
+	if (ret) {
+		LOG_ERR("Failed to initialize INT1");
+		return -EINVAL;
+	}
+
+	ret = bmi270_init_int_pin(&cfg->int2, &data->int2_cb,
+				  bmi270_int2_callback);
+	if (ret) {
+		LOG_ERR("Failed to initialize INT2");
+		return -EINVAL;
+	}
+
+	if (cfg->int1.port) {
+		uint8_t int1_io_ctrl = BMI270_INT_IO_CTRL_OUTPUT_EN;
+		ret = bmi270_reg_write(dev, BMI270_REG_INT1_IO_CTRL, &int1_io_ctrl, 1);
+		if (ret < 0) {
+			LOG_ERR("failed configuring INT1_IO_CTRL (%d)", ret);
+			return ret;
+		}
+	}
+
+	if (cfg->int2.port) {
+		uint8_t int2_io_ctrl = BMI270_INT_IO_CTRL_OUTPUT_EN;
+		ret = bmi270_reg_write(dev, BMI270_REG_INT2_IO_CTRL, &int2_io_ctrl, 1);
+		if (ret < 0) {
+			LOG_ERR("failed configuring INT2_IO_CTRL (%d)", ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+static int bmi270_anymo_config(const struct device *dev, bool enable)
+{
+	const struct bmi270_config *cfg = dev->config;
+	struct bmi270_data *data = dev->data;
+	int ret;
+
+	if (enable) {
+		ret = bmi270_feature_reg_write(dev, cfg->feature->anymo_1,
+					       data->anymo_1);
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	uint16_t anymo_2 = data->anymo_2;
+	if (enable) {
+		anymo_2 |= BMI270_ANYMO_2_ENABLE;
+	}
+
+	ret = bmi270_feature_reg_write(dev, cfg->feature->anymo_2, anymo_2);
+	if (ret < 0) {
+		return ret;
+	}
+
+	uint8_t int1_map_feat = 0;
+
+	if (enable) {
+		int1_map_feat |= BMI270_INT_MAP_ANY_MOTION;
+	}
+
+	ret = bmi270_reg_write(dev, BMI270_REG_INT1_MAP_FEAT, &int1_map_feat, 1);
+	if (ret < 0) {
+		LOG_ERR("failed configuring INT1_MAP_FEAT (%d)", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int bmi270_drdy_config(const struct device *dev, bool enable)
+{
+	int ret;
+
+	uint8_t int_map_data = 0;
+
+	if (enable) {
+		int_map_data |= BMI270_INT_MAP_DATA_DRDY_INT2;
+	}
+
+	ret = bmi270_reg_write(dev, BMI270_REG_INT_MAP_DATA, &int_map_data, 1);
+	if (ret < 0) {
+		LOG_ERR("failed configuring INT_MAP_DATA (%d)", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int bmi270_trigger_set(const struct device *dev,
+		       const struct sensor_trigger *trig,
+		       sensor_trigger_handler_t handler)
+{
+	struct bmi270_data *data = dev->data;
+	const struct bmi270_config *cfg = dev->config;
+
+	switch (trig->type) {
+	case SENSOR_TRIG_MOTION:
+		if (!cfg->int1.port) {
+			return -ENOTSUP;
+		}
+
+		k_mutex_lock(&data->trigger_mutex, K_FOREVER);
+		data->motion_handler = handler;
+		data->motion_trigger = *trig;
```

Comment:
Store the trigger pointer instead of making a copy; this will enable the handler to use `CONTAINER_OF`